Add custom partitioning docs

davidcassany opened the pull request:
This commit adds two new documentation chapters. One covering
custom partitioning scheme in KIWI. And another covering setting
custom volumes in KIWI.

davidcassany commented:
@tomschr I'd really appreciate you could have a look to these new docs for KIWI when you got some time.

tomschr commented on doc/source/building/working_with_images/custom_volumes.rst:
```diff
@@ -0,0 +1,64 @@
+
+Volumes are defined into the KIWI description file, :file:`config.xml`,
+under the `type` element by setting a child `systemdisk` section. The
+`systemdisk` element can include child `volume` elements. 
```
I'd use "child" and "element" always together as in:

```
... can include the `volume` child element.
```

tomschr commented on doc/source/building/working_with_images/custom_volumes.rst:
```diff
@@ -0,0 +1,64 @@
+filesystem supports it, like Btrfs.
+
+Volumes are defined into the KIWI description file, :file:`config.xml`,
+under the `type` element by setting a child `systemdisk` section. The
```
"child" could be removed, making it a bit shorter. But it's a matter of taste.

tomschr commented on doc/source/building/working_with_images/custom_volumes.rst:
```diff
@@ -0,0 +1,64 @@
+under the `type` element by setting a child `systemdisk` section. The
+`systemdisk` element can include child `volume` elements. 
+
+Additional non root volumes are created for each `volume` entry. Volume
```
1. non root -> non-root
1. entry? Is it an element? I would stick with element.

tomschr commented on doc/source/building/working_with_images/custom_volumes.rst:
```diff
@@ -0,0 +1,64 @@
+Additional non root volumes are created for each `volume` entry. Volume
+details can be defined by setting the following `volume` attributes:
+
+  * `name`: This is a required attribute representing volume's name. In
```
1. Remove "This is"
1. volume's name. -> **a** volume's name.

tomschr commented on doc/source/building/working_with_images/custom_volumes.rst:
```diff
@@ -0,0 +1,64 @@
+  * `mountpoint`: Optional attribute that specifies the mountpoint path.
+
+  * `size`: Optional attribute to set the size of the volume. If no suffix
+    (`M` or `G`) is used the value is considered to be in Megabytes.
```
Megabytes -> megabytes

tomschr commented on doc/source/building/working_with_images/custom_volumes.rst:
```diff
@@ -0,0 +1,64 @@
+
+  * `freespace`: Optional attribute to set the additional size added to the
+    volume. If no suffix (`M` or `G`)is used the value is considered to be
+    in Megabytes.
```
Megabytes -> megabytes

davidcassany commented:
> Here are some of my suggestions, ideas, and comments. Sorry if they are so many, most of them are just minor things.

Thanks much, all the feedback is highly appreciated :wink: 

I am addressing your comments now

davidcassany commented on doc/source/building/working_with_images/custom_partitions.rst:
```diff
@@ -0,0 +1,149 @@
+++++++++++++++++++++++++++++++++
+
+Adding additional partitions at boot time of KIWI images is also possible,
+however setting the tools and scripts for doing so falls on behalf of the
```
No it is not a typo, but most probably just a wrong usage of _on behalf of_. I have rephrased it, what about now?

tomschr replied:
Hehe, yes, I stumbled upon also on this *on behalf of*. 

> I have rephrased it, what about now?

Oh, on my GitHub it shows the old version. Also, I only see 1 commit? Would you like to paste the new sentence here?

davidcassany replied:
Oh I did the commit without adding the changes... :P Now should be fine

tomschr replied:
Ahh, now I can see it. Much better! :+1:

davidcassany commented on doc/source/building/working_with_images/custom_partitions.rst:
```diff
@@ -0,0 +1,149 @@
+    creates a new partition. Note that this script should not attempt
+    to create a new partition if it has already created it. 
+
+    .. code:: shell
```
> But inside the shell code, there is no code, only descriptions. So you cheated!

:laughing:  Well I cheated myself, I wanted to write some pseudo-code, but I turned to be just comments.

I like your suggestion, just implemented it.

davidcassany commented on doc/source/building/working_with_images/custom_partitions.rst:
```diff
@@ -0,0 +1,149 @@
+2. Create partitioner shell script matching your specific needs
+
+    Consider the following steps for a partitioner shell script that
+    creates a new partition. Note that this script should not attempt
```
Yes I see your point. What about now?

tomschr replied:
Well done! I like it! :+1:
